File: day1/javascript/solution.js
const { readFileSync } = require("fs");
const input = readFileSync("../input.txt", "utf-8");
const sum = input
    .split('\n')
    .map((line) => {
      const digits = line.match(/\d/g);
      return digits ? parseInt(digits[0] + digits[digits.length - 1]) : 0;
    })
    .reduce((total, value) => total + value, 0);
console.log(sum);

File: day1/python/solution.py
with open("./input.txt", "r") as file:
    input_text = file.read()

import re

data = sum(
    int(digits[0] + digits[-1]) if digits else 0
    for line in input_text.split("\n")
    if (digits := re.findall(r"\d", line))
)

print(data)
